internal/cmd/skupper/listener/kube/listener_create.go

```diff
@@ -0,0 +1,222 @@
+	cmd.CobraCmd.Flags().StringVar(&cmd.flags.tlsSecret, "tls-secret", "", "The name of a Kubernetes secret containing TLS credentials")
+	cmd.CobraCmd.Flags().StringVarP(&cmd.flags.listenerType, "type", "t", "tcp", "The listener type. Choices: [tcp].")
+	cmd.CobraCmd.Flags().StringVarP(&cmd.flags.output, "output", "o", "", "print resources to the console instead of submitting them to the Skupper controller. Choices: json, yaml")
+}
+
+func (cmd *CmdListenerCreate) ValidateInput(args []string) []error {
+	var validationErrors []error
+	resourceStringValidator := validator.NewResourceStringValidator()
+	numberValidator := validator.NewNumberValidator()
+	listenerTypeValidator := validator.NewOptionValidator(utils.ListenerTypes)
+	outputTypeValidator := validator.NewOptionValidator(utils.OutputTypes)
+
+	// Validate arguments name and port
+	if len(args) < 2 {
+		validationErrors = append(validationErrors, fmt.Errorf("listener name and port must be configured"))
+	} else if len(args) > 2 {
+		validationErrors = append(validationErrors, fmt.Errorf("only two arguments are allowed for this command"))
+	} else if args[0] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("listener name must not be empty"))
+	} else if args[1] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("listener port must not be empty"))
+	} else {
+		cmd.name = args[0]
+		ok, err := resourceStringValidator.Evaluate(cmd.name)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("listener name is not valid: %s", err))
+		}
+
+		cmd.port, err = strconv.Atoi(args[1])
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("listener port is not valid: %s", err))
+		}
+		ok, err = numberValidator.Evaluate(cmd.port)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("listener port is not valid: %s", err))
+		}
+	}
+
+	// Validate if there is already a listener with this name in the namespace
+	if cmd.name != "" {
+		listener, err := cmd.client.Listeners(cmd.namespace).Get(context.TODO(), cmd.name, metav1.GetOptions{})
+		if listener != nil && !errors.IsNotFound(err) && listener.Status.StatusMessage == "Ok" {
+			validationErrors = append(validationErrors, fmt.Errorf("there is already a listener %s created for namespace %s", cmd.name, cmd.namespace))
+		}
+	}
+
+	// Validate flags
+	if cmd.flags.routingKey != "" {
+		//TBD what characters are not allowed
+		ok, err := resourceStringValidator.Evaluate(cmd.flags.routingKey)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("routing key is not valid: %s", err))
+		}
+	}
+	if cmd.flags.host != "" {
+		//TBD what characters are not allowed
+		ok, err := resourceStringValidator.Evaluate(cmd.flags.host)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("host name is not valid: %s", err))
+		}
+	}
+	if cmd.flags.tlsSecret != "" {
+		//TBD what characters are allowed
```

Comment:
This comment (line 142) can be removed as well. We need to validate the data that is going to be persisted, but in this case, the listener is not going to be created at all if the name of the secret does not exist in the kubernetes namespace.

Reply:
fixed